tests/suite/test_ts_tls_passthrough.py

```diff
@@ -0,0 +1,225 @@
+                    "-enable-leader-election=false",
+                    "-enable-tls-passthrough=true",
+                ],
+            },
+            {"example": "transport-server-tls-passthrough"},
+        )
+    ],
+    indirect=True,
+)
+class TestTransportServerTlsPassthrough:
+    def restore_ts(self, kube_apis, transport_server_tls_passthrough_setup) -> None:
+        """
+        Function to revert a TransportServer resource to a valid state.
+        """
+        ts_std_src = f"{TEST_DATA}/transport-server-tls-passthrough/standard/transport-server.yaml"
+        ts_std_res = create_ts_from_yaml(
+                        kube_apis.custom_objects,
+                        ts_std_src,
+                        transport_server_tls_passthrough_setup.namespace,
+                    )
+        wait_before_test(1)
+        pprint(ts_std_res)
+
+    @pytest.mark.smoke
+    def test_tls_passthrough(
+        self,
+        kube_apis,
+        crd_ingress_controller,
+        transport_server_tls_passthrough_setup,
+        test_namespace,
+    ):
+        """
+            Test TransportServer TLS passthrough on https port.
+        """
+        session = create_sni_session()
+        req_url = (
+            f"https://{transport_server_tls_passthrough_setup.public_endpoint.public_ip}:"
+            f"{transport_server_tls_passthrough_setup.public_endpoint.port_ssl}"
+        )
+        wait_before_test()
+        resp = session.get(
+            req_url,
+            headers={"host": transport_server_tls_passthrough_setup.ts_host},
+            verify=False,
+        )
+        assert resp.status_code == 200
+        assert f"hello from pod {get_first_pod_name(kube_apis.v1, test_namespace)}" in resp.text
+
+    def test_tls_passthrough_host_collision_ts(
+        self,
+        kube_apis,
+        crd_ingress_controller,
+        transport_server_tls_passthrough_setup,
+        test_namespace,
+    ):
+        """
+            Test host collision handling in TransportServer with another TransportServer.
+        """
+        print("Step 1: Create second TS with same host")
+        ts_src_same_host = (
+            f"{TEST_DATA}/transport-server-tls-passthrough/transport-server-same-host.yaml"
+        )
+        ts_same_host = create_ts_from_yaml(
+            kube_apis.custom_objects, ts_src_same_host, test_namespace
+        )
+        wait_before_test()
+        response = read_ts(
+            kube_apis.custom_objects, test_namespace, ts_same_host["metadata"]["name"]
+        )
+        assert (
+            response["status"]["reason"] == "Rejected"
+            and response["status"]["message"] == "Host is taken by another resource"
+        )
+
+        print("Step 2: Delete TS taking up the host")
+        delete_ts(
+            kube_apis.custom_objects,
+            transport_server_tls_passthrough_setup.ts_resource,
+            test_namespace,
+        )
+        wait_before_test(1)
```

Comment:
shame about these sleep commands, do we have any `waitUntil(condition, timeout)` style functions available ?